### PR TITLE
Topics targets messages

### PR DESCRIPTION
**Pull request by gewenyu99**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: src/routes/docs/products/messaging/messages/+page.markdoc [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+* optional
+* An ISO date time string specifying when the SMS should be sent. This parameter must be set if the status is `scheduled` and must represent a future time.
+{% /table %}
+{% /tabsitem %}
+{% /tabs %}
+
+# Sending a message {% #create-a-message %}
+You can create a message with the `METHOD` method using a Server SDK.
+{% multicode %}
+```js
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/messages/+page.markdoc:207]: @abnegate @stnguyen90 Simple code example of just the three methods.

We will cover full flow in detail in their journies.

File: src/routes/docs/products/messaging/targets/+page.markdoc [hunk 2]
```diff
@@ -4,4 +4,20 @@ title: Targets
...
+A user can have multiple targets, such as emails, phone numbers, and devices with your app installed. 
+These targets can subscribe to a topic, so when messages are published to a topic, all subscribed targets receive the message.
+
+{% arrow_link href="/docs/products/messaging/topics" %}
+Learn more about topics
+{% /arrow_link %}
+
+# Types of targets {% #types-of-targets %}
+
+[@abnegate @stnguyen90: TODO types of targets and how to create/add targets in console + code.]
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/targets/+page.markdoc:21]: @abnegate @stnguyen90 What are the different target types called?

What's the code like for creating targets?

Is there client code required?

File: src/routes/docs/products/messaging/topics/+page.markdoc [hunk 3]
```diff
@@ -1,7 +1,57 @@
...
+# Create a topic {% #create-a-topic %}
+You can create topics 
+{% tabs %}
+{% tabsitem #console title="Console" %}
+[SCREENSHOT]
+Navigate to your Appwrite Console > **Messaging** > **Topics** > **Create topic**.
+{% /tabsitem %}
+
+{% tabsitem #server-sdk title="Server SDK" %}
+Lorem ipsum dolor sit amet consectetur.
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/topics/+page.markdoc:36]: @abnegate @stnguyen90 Simple server SDK examples <3

File: src/routes/docs/products/messaging/topics/+page.markdoc [hunk 4]
```diff
@@ -1,7 +1,57 @@
...
+{% tabs %}
+{% tabsitem #console title="Console" %}
+[SCREENSHOT]
+Navigate to your Appwrite Console > **Messaging** > **Topics** > click on your topic > **Subscribers** > **Create topic** > **Add subscriber**.
+
+If you can't find the targets you'd like to add, see the [targets page](/docs/products/messaging/targets).
+{% /tabsitem %}
+
+{% tabsitem #server-sdk title="Server SDK" %}
+Lorem ipsum dolor sit amet consectetur.
```
Comment (gewenyu99) [inline src/routes/docs/products/messaging/topics/+page.markdoc:54]: @abnegate @stnguyen90 Simple server SDK examples <3

**Review by gewenyu99 — COMMENTED** [review]
Start of concept pages

File: src/routes/docs/products/messaging/targets/+page.markdoc (hunk 2 above)
Comment (stnguyen90) [reply src/routes/docs/products/messaging/targets/+page.markdoc:21]: > What are the different target types called?

```php
// Message types
const MESSAGE_TYPE_EMAIL = 'email';
const MESSAGE_TYPE_SMS = 'sms';
const MESSAGE_TYPE_PUSH = 'push';
```

https://github.com/appwrite/appwrite/blob/49a753509b17ee4209999e08678e8cad71a20bee/app/init.php#L197-L199

> What's the code like for creating targets?

[Client-side](https://github.com/appwrite/appwrite/blob/49a753509b17ee4209999e08678e8cad71a20bee/app/controllers/api/account.php#L1951-L1972), you can only create a push target:

```
account.createPushTarget()
```

[Server-side](https://github.com/appwrite/appwrite/blob/49a753509b17ee4209999e08678e8cad71a20bee/app/controllers/api/users.php#L429-L452), you can create any target:

```
users.createTarget()
```

> Is there client code required?

Did I answer your question?

File: src/routes/docs/products/messaging/messages/+page.markdoc (hunk 1 above)
Comment (stnguyen90) [reply src/routes/docs/products/messaging/messages/+page.markdoc:207]: You should be able to determine the methods from the [API](https://github.com/appwrite/appwrite/blob/49a753509b17ee4209999e08678e8cad71a20bee/app/controllers/api/messaging.php#L2275-L2306). Do you need additional detail?

**Review by stnguyen90 — COMMENTED** [review]
see comments

**Pull request merged**